L/libpolymake_julia/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+import Pkg: PackageSpec
+
+const name = "libpolymake_julia"
+const version = v"0.1.0"
+
+# Collection of sources required to build libpolymake_julia
+const sources = [
+    ArchiveSource("https://github.com/oscar-system/libpolymake-julia/archive/v$(version).tar.gz",
+                  "533237a1a1f7ff544017b9576704b2f05c6e3139443ea0aaa20bc88560675e9a"),
+]
+
+# Bash recipe for building across all platforms
+const script = raw"""
+# remove $libdir from LD_LIBRARY_PATH as this causes issues with perl
+if [[ -n "$LD_LIBRARY_PATH" ]]; then
+LD_LIBRARY_PATH=$(echo -n $LD_LIBRARY_PATH | sed -e "s|[:^]$libdir\w*|:|g")
+fi
+
+cmake libpolymake-j*/ -B build \
+   -DJulia_PREFIX="$prefix" \
+   -DCMAKE_INSTALL_PREFIX="$prefix" \
+   -DCMAKE_FIND_ROOT_PATH="$prefix" \
+   -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
+   -DCMAKE_BUILD_TYPE=Release
+
+VERBOSE=ON cmake --build build --config Release --target install -- -j${nproc}
+
+install_license $WORKSPACE/srcdir/libpolymake-j*/LICENSE.md
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+const platforms = expand_cxxstring_abis([
+    Linux(:x86_64; libc=:glibc, compiler_abi=CompilerABI(cxxstring_abi=:cxx11)),
+    MacOS(:x86_64; compiler_abi=CompilerABI(cxxstring_abi=:cxx11)),
+])
+
+# The products that we will ensure are always built
+const products = [
+    ExecutableProduct("polymake_run_script", :polymake_run_script),
+    LibraryProduct("libpolymake_julia", :libpolymake_julia),
+    FileProduct("share/libpolymake_julia/type_translator.jl",:type_translator),
+]
+
+# Dependencies that must be installed before this package can be built
+const dependencies = [
+    Dependency("CompilerSupportLibraries_jll"),
+    BuildDependency(PackageSpec(name="Julia_jll", version="v1.4.1")),
+    Dependency("libcxxwrap_julia_jll"),
```

Comment:
Perhaps this should specify a version of `libcxxwrap_julia`? Otherwise, couldn't a user endup with a `libpolymake_julia` built against, say, `libcxxwrap_julia` 0.8 but then during runtime `libcxxwrap_julia` 0.9 is installed (or something like that), potentially leading to havoc?

I am wondering the same for `polymake_jll`

Reply:
I'm not totally convinced as this would create very strict compat entries, e.g. `libcxxwrap_julia =0.8.0` and require rebuilding for any patch version (If I understand all that correctly).

Restricting the versions using the `Project.toml` of `Polymake.jl` allows more flexibility and it keeps the restrictions in one place and not spread out over various repositories. Also this is the place where we can experiment and run tests with various versions.
The tests in the `Polymake.jl` pull request indicate that using 0.8.0 at build time but using 0.7.1 at runtime does work.

Reply:
Oh, sure, we shouldn't fix a single version like 0.8.0, fully agreed. But we could use something like this:
```suggestion
    Dependency(PackageSpec(name="libcxxwrap_julia_jll", version=Pkg.Types.VersionSpec("0.7.1-0.8")),
```
Anyway, I am also OK with leaving this unrestricted right now.

Reply:
I didn't know that this was possible, I only noticed that using `v0.8` would produce `=0.8.0` in some jll repositories but couldn't find any other examples.
I still prefer to have the restrictions where I can run tests, i.e. in Polymake.jl.